bin/build-locales

```diff
@@ -64,66 +64,82 @@ try {
     fs.mkdirSync(outputDir);
 }
 
-// message key with english string values (i.e. default values)
-var viewLocales = {};
-// FormattedMessage id with english string as value. Use for default values in translations
-// Sample structure: { 'general-general.blah': 'blah', 'about-about.blah': 'blahblah' }
-var idsWithICU = {};
-// reverse (i.e. english string with message key as the value) object for searching po files.
-// Sample structure: { 'blah': 'general-general.blah', 'blahblah': 'about-about.blah' }
-var icuWithIds = {};
-
 // get global locale strings first.
 var globalTemplateFile = path.resolve(__dirname, '../src/l10n.json');
-localeCompare.getIdsForView('general', globalTemplateFile, viewLocales, idsWithICU, icuWithIds);
+var ids = JSON.parse(fs.readFileSync(globalTemplateFile, 'utf8'));
 
+// message key with english string values (i.e. default values)
+var viewLocales = {
+    general: {
+        en: ids
+    }
+};
+var idsWithICU = localeCompare.idToICUMap('general', ids);
+var icuWithIds = localeCompare.icuToIdMap('general', ids);
+// localeCompare.getIdsForView('general', globalTemplateFile, viewLocales, idsWithICU, icuWithIds);
 
-// start with all views, and remove localized ones as they are iterated over
-var views = glob.sync(path.resolve(__dirname, '../src/views/*'));
-for (var i = 0; i < views.length; i++) {
-    views[i] = views[i].split('/').pop();
-}
+var views = [];
+var localizedAssetUrls = {};
 
-// get view-specific locale strings.
-var files = glob.sync(path.resolve(__dirname, '../src/views/**/l10n.json'));
-files.forEach(function (file) {
-    var dirPath = file.split('/');
-    dirPath.pop();
-    var view = dirPath.pop();
-    localeCompare.getIdsForView(view, file, viewLocales, idsWithICU, icuWithIds);
-});
+// start with all views, and remove localized ones as they are iterated over
+for (var v in routes) {
+    if (typeof routes[v].redirect !== 'undefined') {
+        continue;
+    }
+    
+    views.push(routes[v].name);
+    try {
+        var subdir = routes[v].view.split('/');
+        subdir.pop();
+        var l10n = path.resolve(__dirname, '../src/views/' + subdir.join('/') + '/l10n.json');
+        fs.accessSync(l10n, fs.F_OK);
+        ids = JSON.parse(fs.readFileSync(l10n, 'utf8'));
+        viewLocales[routes[v].name] = {
+            en: ids
+        };
+        idsWithICU = merge(idsWithICU, localeCompare.idToICUMap(routes[v].name, ids));
+        icuWithIds = merge(icuWithIds, localeCompare.icuToIdMap(routes[v].name, ids));
+    } catch (err) {
+        // check that routes' view path is not malformed, error if it is
+        try {
+            fs.accessSync(path.resolve(__dirname, '../src/views/' + routes[v].view + '.jsx'));
+        } catch (err) {
+            // the config for the view is not set up correctly, so throw the error.
+            throw err;
```

Comment:
Is this handled this way to silence errors for nonexistent l10n.json files? If so, I think it'd be more appropriate to explicitly check if the file exists so that anything else that goes wrong will get thrown and we'll see it.